website/src/pages/plugins/presets/preset-client.mdx

```diff
@@ -58,6 +58,7 @@ The `client` preset allows the following `config` options:
 - [`nonOptionalTypename`](/plugins/typescript/typescript#nonoptionaltypename): Automatically adds `__typename` field to the generated types, even when they are not specified in the selection set, and makes it non-optional.
 - [`avoidOptionals`](/plugins/typescript/typescript#avoidoptionals): This will cause the generator to avoid using TypeScript optionals (`?`) on types.
 - [`documentMode`](#documentmode): Allows you to control how the documents are generated.
+- [`semanticNonNull`](/plugins/typescript/typescript-operations#semanticnonnull): Handles `@semanticNonNull` directive based on client-side requirements.
```

Comment:
```suggestion
- [`semanticNullability`](/plugins/typescript/typescript-operations#semanticnullability): Indicate the client capabilities to get stronger types with [semantic nullability](https://github.com/graphql/graphql-wg/blob/main/rfcs/SemanticNullability.md)-enabled schemas.
```

Reply:
By using `graphql-sock` this supports not only `@semanticNonNull` the directive but all the proposed solutions (assuming GraphQL.js support) to the semantic nullability RFC. Theoretically this should already work just fine with the `*` syntax when using `graphql@canary-pr-4192` for example (I've not tested it though). If we were to come up with an alternative solution (e.g. `GraphQLStrictNonNull` to represent the old "kills parent on exception" style of non-null) then "semantic nullability" would still cover this solution, but "semanticNonNull" would not.

Reply:
I'd support renaming the option to simply `nullability` though, as you proposed separately.